endpoints/publicProxy/crypto.go

```diff
@@ -0,0 +1,57 @@
+package publicProxy
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/go-jose/go-jose/v4"
+	"golang.org/x/crypto/hkdf"
+)
+
+// deriveKey uses HKDF to expand a "password" into a []byte to be used as a key; better than just a raw hash
+func deriveKey(keyString string, sz int) ([]byte, error) {
```

Comment:
Recommend salting the KDF with a stable, context-specific, non-secret value like the client_id from the client credentials flow or database index of the provider.